src/envoy/http/service_control/filter.cc

```diff
@@ -49,7 +49,7 @@ Envoy::Http::FilterHeadersStatus ServiceControlFilter::decodeHeaders(
 
   handler_ =
       factory_.createHandler(headers, decoder_callbacks_->streamInfo(), stats_);
```

Comment:
There is another createHandler in the ::log() function.  we need to add there too.  Or just pass filterState to createHandler()

Reply:
It is unlikely you can pass changeable filterState to createHandler since it is from `const` streamInfo.
We only add filterState once in decodeHeader.

Reply:
Can we just remove the `createHandler` in `log` and assume it is always set? Envoy will always guarantee to call `decodeHeaders` before `log`.

Reply:
Yes, I believe the accessLog from HCM is collected earlier than that of listeners.
https://github.com/envoyproxy/envoy/blob/88dcb292817946510bb87f8f379a954962cece79/source/common/http/conn_manager_impl.cc#L618

Reply:
It is just protecting from nullness. No much value to remove it.